content/articles/The-concept-of-cookies-in-servlets/index.md

```diff
@@ -0,0 +1,222 @@
+
+### How does Cookie work
+When a client makes a new request to a Web Server, the web server assigns a unique ID as a `cookie`, which is then sent back with the response to the client and saved in the client's browser.
+
+This `cookie` value is likewise provided to the server when the client sends another request, and the server uses this value to identify the user.
+
+In this case, a cookie is used to store information required each time the client visits a website, such as a user ID, name, or other identifiers.
+### Types of Cookies in Java Servlet
+Two types of cookies in the servlet are Non-persistent and persistent cookies.
+
+#### Non persistent cookies
+
+A non-persistent/Session cookie is only valid for one session and is automatically deleted when the browser is closed.
+
+#### persistent cookies
+
+It's valid for multiple sessions and, unlike non-persistent cookies, it's not deleted when the user closes the browser; instead, it's deleted only when the user logs out or signs out.
+
+### Advantage of Cookies
+
+- Transparent: Cookies work in this case without the user being aware that data is being saved in their browser.
+- Simplest technique: Cookies are the most basic method for keeping client states while connected to a web server.
+- Because cookies are stored in the client browser, they reduce the load on the server's memory.
+### Disadvantage 
+-  Cookies will only work if the user's web browser allows them; otherwise, they will not.
+- In a Cookie object, only textual data can be stored.
+
+### Cookie class and the methods required for cookies
+
+#### Cookie class
+ A `Cookie`, is a class that contains several cookie-related methods that allow you to use cookies.
+#### Methods of Cookie class
+- `void setMaxAge()`: For example, this method can be used to set Cookie's maximum age. The cookie lifetime can be specified when max-age is set in web applications.
+- `void setValue()`: When a new value is assigned to the cookie, this method is used.
+- `String getName()`: Returns the cookie's name, which cannot be modified after it has been created.
+- `String getValue()`: This method aids in obtaining the cookie's current value.
+- `int getVersion()`: Cookie's version number is returned by this method.
+- `void setVersion()`: Cookie's version number is set in this method.
+- ` int getMaxAge()`: This method is used to determine a cookie's maximum age.
+- `String getPath()`: The path from the cookie is obtained using this method.
+- `void setPath()`: This method is used to specify a cookie path. In a web application, for example, it is possible to specify a specific path for the cookies to be saved.

```

Comment:
Change to "- `void setPath()`: This method is used to specify a cookie path. For example, in a web application, it is possible to specify a specific path for the cookies to be saved."